pkg/proxy/proxy.go

```diff
@@ -488,7 +488,13 @@ func (p *Proxy) CreateOrUpdateRedirect(l4 *policy.L4Filter, id string, source Pr
 		return nil, err
 	}
 
-	if strings.ToLower(l4.L7Parser) != "http" {
+	switch policy.L7ParserType(l4.L7Parser) {
+	case policy.ParserTypeHTTP:
+	case policy.ParserTypeKafka:
+		// TODO We need to remove this once Kakfa parser and router support is in.
+		log.Debug("Kafka Parser not supported yet")
+		return nil, fmt.Errorf("unsupported L7 protocol proxy:\"%s\"", l4.L7Parser)
```

Comment:
2 minor nitpicks:
`\"%s\"` should be replaced with `%q`
Add a space after the semi-colon `:`.